testing/btest/Baseline/core.tunnels.gre-over-udp/conn.log

```diff
@@ -0,0 +1,14 @@
+### BTest baseline data generated by btest-diff. Do not edit. Use "btest -U/-u" to update. Requires BTest >= 0.63.
+#separator \x09
+#set_separator	,
+#empty_field	(empty)
+#unset_field	-
+#path	conn
+#open XXXX-XX-XX-XX-XX-XX
+#fields	ts	uid	id.orig_h	id.orig_p	id.resp_h	id.resp_p	proto	service	duration	orig_bytes	resp_bytes	conn_state	local_orig	local_resp	missed_bytes	history	orig_pkts	orig_ip_bytes	resp_pkts	resp_ip_bytes	tunnel_parents
+#types	time	string	addr	port	addr	port	enum	string	interval	count	count	string	bool	bool	count	string	count	count	count	count	set[string]
+XXXXXXXXXX.XXXXXX	C4J4Th3PJpwUYZZ6gc	172.17.0.5	45690	1.1.1.1	53	udp	dns	0.000158	52	0	S0	T	F	0	D	2	108	0	0	ClEkJM2Vm5giqnMf4h
+XXXXXXXXXX.XXXXXX	CUM0KZ3MLUfNB0cl11	172.17.0.5	47478	192.0.78.150	80	tcp	http	0.090287	72	0	SH	T	F	0	SADF	6	332	0	0	ClEkJM2Vm5giqnMf4h
+XXXXXXXXXX.XXXXXX	CHhAvVGS1DHFjwGM9	192.168.0.107	48282	192.168.5.1	4754	udp	-	0.000158	116	0	S0	T	T	0	D	2	172	0	0	-
+XXXXXXXXXX.XXXXXX	CtPZjS20MLrsMUOJi2	192.168.0.107	49714	192.168.5.1	4754	udp	-	0.090287	356	0	S0	T	T	0	D	6	524	0	0	-
```

Comment:
There's something off with this :-)

The tunnel_parent `ClEkJM2Vm5giqnMf4h` is used for *both* inner connections. If you look at the pcap, however, the inner DNS connection is tunneled within the `orig_p=48282` connection, while HTTP is tunneled within the `orig_p=49714` connection (src_port chosen as hash of inner flow).

![Screenshot-20231012-091409](https://github.com/zeek/zeek/assets/2100509/7430ac0b-023d-460b-8e68-47608cfe0c39)

![Screenshot-20231012-090918](https://github.com/zeek/zeek/assets/2100509/43452dca-01f4-42e2-a98d-997435e2b8ac)

So, there are two tunnel connections, but we only show a single one in `tunnel.log`. What is more, that `tunnel.log` uid isn't actually in the `conn.log`.

This is probably a more fundamental limitation around the current IP/GRE tunneling logic that needs to be made aware of the underlying connection.

I'd be okay to merge as is, but we should create a follow-up to straighten this out. Expectation would be two tunnel.log entries and two distinct tunnel_parents in the conn.log.

Reply:
I don't think this is related to the change in this PR. I agree we should move this to a new issue and fix it separately.

Reply:
For a GRE over UDP tunnel where the source port is a hash of the inner flow, it actually would be nice if there was only a single tunnel log entry, and not one for every single inner connection.  After all, there is actually only one tunnel.  I've seen this issue a bit with vxlan, where every conn log entry also has a corresponding vxlan and tunnel log, it's not ideal.

Reply:
Yes, but with the current model it would be more correct to have the individual outer connections. I do like the thought of removing the src port from the connection's tuple for these tunnels, but it would be quite different behavior.